### ISSUE TITLE
Apps with changed icon color won't launch

### ISSUE THREAD

**Issue by Yquas** [markusfisch/PieLauncher#125]
Some apps like "Tasks.org", "Simple Contacts" or "Simple Calendar" give users the ability to change the color of the app's icon in settings. Pie launcher can only launch these apps if their icon color is kept on the default color, or else it will only play the launch animation but won't launch the app, forcing one to launch these apps in some other way.

**Comment by markusfisch** [comment]
Thanks a lot for reporting! 👍  I wasn't aware of this bug.

**Comment by chesio** [comment]
I have multiple Fossify Apps (fork of Simple Apps) installed and have custom color set to every one of them and I can launch the apps without problem (from pie menu and from apps menu as well...)

**Comment by Yquas** [comment]
Okay, if it works for both of you then the problem could be my device - it runs MIUI 11 / Android 10.

I get both "Tasks.org" and "Simple Mobile Tools" from F-Droid. After seeing your comments I also tested "Fossify Apps" from both the Play Store and from F-Droid but they all behave in exactly the same way: On the default icon color all apps launch, when changed to a non-default color only the launch animation plays, and when changed back to the default color all apps launch again.
Is there any way for me to give you some kind of technical bug report? I don't know what else would help when it works fine for you.

**Comment by markusfisch** [comment]
Well, you could check the device logs when this is happening with `logcat`, if you have the `adb` command line tool (from the Android SDK, then read [here](https://developer.android.com/tools/logcat) how to check the logs) or [Android Studio](https://developer.android.com/studio) installed (then follow this [guide](https://developer.android.com/studio/debug/logcat)). Hopefully there's an error message when launching an affected app fails.

Then, if you like, and if you're curious enough, you could also clone this repository and build and run the app from source on your device with Android Studio. It's really quite easy to do. Check this [YouTube](https://www.youtube.com/watch?v=fQjA1CwV8lE) guide for example. The app is built and installed in a debug version so it can be easily installed beside the production app.

Then, this [code snippet](https://github.com/markusfisch/PieLauncher/blob/master/app/src/main/java/de/markusfisch/android/pielauncher/content/AppMenu.java#L120) launches apps in Pie Launcher. You could remove the `try/catch` block to see (in the logs) if there are any exceptions (which would crash the app).

**Comment by Sergueille** [comment]
I have the same problem with `com.textra`.
I tried what you said, and there is no exception thrown by the snippet you given.
However, if I put `startMainActivity` out of the `if (lm.isActivityEnabled(icon.componentName, icon.userHandle))`, the app launches just fine.

**Comment by Sergueille** [comment]
And here is more info: 
* `icon.componentName.toString()` is `ComponentInfo{com.textra/com.mplus.lib.Main_ff2e7d32}` (looks strange?),
* `icon.componentName.getPackageName()` is `com.textra`
* `icon.userHandle.toString()` is `UserHandle{0}`

**Comment by Sergueille** [comment]
Hello again @markusfisch, I think I have a clue what causes the bug. It seems that [this check](https://github.com/markusfisch/PieLauncher/blob/master/app/src/main/java/de/markusfisch/android/pielauncher/content/AppMenu.java#L126) doesn't pass when it should, because `icon.componentName` is not the expected value due to the changed color of the app. However, if I replace `isActivityEnabled` with `isPackageEnabled`, the app is launched, but I can't test completely if it works as intended because my device doesn't support multiple user profiles.

**Comment by markusfisch** [comment]
@Sergueille Good catch! This may indeed be the reason! I'm currently reworking this section.

**Comment by markusfisch** [comment]
Sorry that it didn't work, and thanks a lot for reporting that the bug still exists. Very much appreciated!

At least the new Activity not enabled message helps to narrow the problem down.

**Comment by Sergueille** [comment]
When the color is changed,  `icon.componentName` looks like `"app.mainActivity.color"`, and the only activity returned by `launcherApps.getActivityList` is this activity. Then, `launcherApps.isActivityEnabled` returns `false` and the toast is shown.

I don't quite understand why these check are being made, would it be more reliable to instead launch the activity directly and then catch exceptions?

**Comment by markusfisch** [comment]
Good point! Currently `isActivityEnabled` is used to find an enabled activity if the original one got disabled. But that doesn't seem to work as expected, so catching the exception may also be a good approach. Will try.
